sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -385,9 +385,15 @@ object DateTimeUtils {
    * microseconds where microsecond 0 is 1970-01-01 00:00:00Z.
    */
   def instantToMicros(instant: Instant): Long = {
-    val us = Math.multiplyExact(instant.getEpochSecond, MICROS_PER_SECOND)
-    val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
-    result
+    if (instant.getEpochSecond < 0) {
```

Comment:
@dgd-contributor Could you follow the approach from:
https://github.com/apache/spark/blob/2c8ced95905b8a9f7b98c4913a385d25da455e7a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala#L872-L885
and handle especially just small range. So, perf impact should be smaller.

Reply:
Ok done